### PR TITLE
[Activation] Signal on block disconnected from CM

### LINKED ISSUE #1814: Signal on block disconnected in ConsensusManager
Signal on block disconnected in ConsensusManager when block is disconnected and use new Rewind method that accepts blocks that are being disconnected.

https://github.com/stratisproject/StratisBitcoinFullNode/issues/1768

### PR DESCRIPTION
**Pull request by noescape00**
Fix #1814

### CODE REVIEW

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 1]
```diff
@@ -591,11 +579,21 @@ private async Task RewindToForkPointAsync(ChainedHeader fork, ChainedHeader oldT
...
+            while (current != fork)
             {
                 await this.consensusRules.RewindAsync().ConfigureAwait(false);
+
+                lock (this.peerLock)
+                {
+                    this.SetConsensusTipInternalLocked(current.Previous);
+                }
+
+                // Signal disconnected block.
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: Don't need this comment

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 2]
```diff
@@ -610,11 +608,17 @@ private async Task RewindPartiallyConnectedChainAsync(ChainedHeader lastValidate
...
-                RewindState transitionState = await this.consensusRules.RewindAsync().ConfigureAwait(false);
+                await this.consensusRules.RewindAsync().ConfigureAwait(false);
+
                 lock (this.peerLock)
                 {
-                    current = this.chainedHeaderTree.GetChainedHeader(transitionState.BlockHash);
+                    this.SetConsensusTipInternalLocked(current.Previous);
                 }
+
+                // Signal disconnected block.
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: Don't need this comment

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 3]
```diff
@@ -610,11 +608,17 @@ private async Task RewindPartiallyConnectedChainAsync(ChainedHeader lastValidate
 
             while (currentTip.Height < current.Height)
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: This method looks to me that it is exactly the same as the method above, can't we just use one method and deleted the other one?

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs (hunk 1 above)
Comment (noescape00) [reply src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: What did you do with real Aprogiena?

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs (hunk 2 above)
Comment (noescape00) [reply src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: fixed

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs (hunk 1 above)
Comment (noescape00) [reply src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: fixed

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs (hunk 3 above)
Comment (noescape00) [reply src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: fixed

**Pull request merged**